packages/inertia-vue/src/link.js

```diff
@@ -67,7 +67,7 @@ export default {
               method: props.method,
               replace: props.replace,
               preserveScroll: props.preserveScroll,
-              preserveState: props.preserveState,
+              preserveState: props.preserveState ?? (props.method !== 'get'),
```

Comment:
Since this depends on user input, people could pass `GET` and experience unexpected behaviour

```suggestion
              preserveState: props.preserveState ?? (props.method.toLowerCase() !== 'get'),
```